backend/api/Services/DocumentService.cs

```diff
@@ -95,8 +96,17 @@ public async Task<DocumentUploadResponse> UploadActivityDocumentAsync(long activ
             DocumentUploadResponse response = new DocumentUploadResponse() { ExternalResult = externalResult };
             if (externalResult.Status == ExternalResultStatus.Success)
             {
-                // Create the pims document
                 var externalDocument = externalResult.Payload;
+
+                // Save metadata of document
+                IList<Task<ExternalResult<DocumentMetadata>>> metadataTasks = new List<Task<ExternalResult<DocumentMetadata>>>();
+                foreach (var metadata in uploadRequest.DocumentMetadata)
```

Comment:
I'm not too familiar with this api, but I take it that there is no way of sending the metadata in a batched api call, or with the document itself? There shouldn't be too many metadata fields so I don't think this is a big deal, but still could result in a fair number of rest requests.

Reply:
looked into swagger but there is no batch request, we need to send it one by one.